MaxText/layers/linears.py

```diff
@@ -686,25 +734,29 @@ def dense_matmul(self, inputs, gate_logits, w0_kernel, w1_kernel, wo_kernel):
         wo_kernel_axes = ("exp", "mlp", None)
         wo_kernel = self.maybe_all_gather_kernel_weight_in_expert_parallelism(wo_kernel, wo_kernel_axes)
         intermediate_layer = self.get_einsum(rhs_mesh_axes=wo_kernel_axes)(
-            "EBCH,EHM -> EBCM", layer_multiply, wo_kernel, precision=matmul_precision
-        )
-        intermediate_layer = nn.with_logical_constraint(
-            intermediate_layer,
-            ("activation_exp", "activation_batch_no_exp", None, "activation_embed"),
+            "EBNCH,EHM -> EBNCM", layer_multiply, wo_kernel, precision=matmul_precision
         )
         if self.config.activations_in_float32:
           intermediate_layer = intermediate_layer.astype(jnp.float32)
+        # intermediate_layer = nn.with_logical_constraint(
```

Comment:
Is this sharding config affect inference?

Reply:
yes, I added condition that only affect inference.